src/ast/passes/semantic_analyser.cpp

```diff
@@ -574,6 +574,55 @@ static const std::map<std::string, call_spec> CALL_SPEC = {
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::integer, .literal=true } } } },
+  { "strcmp",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string } } } },
+  { "strnlen",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::integer } } } },
+  { "strlen",
+    { .min_args=1,
+      .max_args=1,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string } } } },
+  { "strspn",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string } } } },
+  { "strcspn",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string } } } },
+  { "strnstr",
+    { .min_args=3,
+      .max_args=3,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::integer, .literal=true } } } },
+  { "strstr",
```

Comment:
We already have `strcontains` which does practically the same thing. We should call `bpf_strstr` (or `bpf_strnstr`) from there.

Reply:
I just try `strcontaines` use bpf_strstr first in PR (https://github.com/bpftrace/bpftrace/pull/4532), please review.